packages/commons/src/rate-limiter/rateLimiterMiddleware.ts

```diff
@@ -17,18 +19,25 @@ export function rateLimiterMiddleware(
   return async (req, res, next) => {
     const ctx = fromAppContext(req.ctx);
 
-    if (!ctx.authData?.organizationId) {
+    const organizationId: TenantId | undefined = getUserInfoFromAuthData(
+      ctx.authData
+    ).organizationId;
+
+    if (!organizationId) {
       const errorRes = makeApiProblem(
-        genericError("Missing expected organizationId claim in token"),
-        () => constants.HTTP_STATUS_INTERNAL_SERVER_ERROR,
+        unauthorizedError(
+          `No organizationId found in authData -- cannot apply rate limiting`
+        ),
+        () => constants.HTTP_STATUS_UNAUTHORIZED,
```

Comment:
few points here:
- generally I would not expect a rate limit middleware to return unauthorized error
- since this is unexpected, maybe it's better to keep returning internal server error to receive the alarm

Reply:
True, not sure why I changed it from before :) Done